routers/web/auth/auth.go

```diff
@@ -137,6 +137,32 @@ func checkAutoLogin(ctx *context.Context) bool {
 	return false
 }
 
+func checkForceOAuth(ctx *context.Context) bool {
+	// Check if authentication is forced to OAuth
+
+	authSources, err := auth.GetActiveOAuth2ProviderSources()
+	if err != nil {
+		return false
+	}
+
+	var OAuthList []int64
+
+	for _, source := range authSources {
+		if forced, ok := source.Cfg.(auth_service.ForceOAuth); ok && forced.IsOAuthForced() {
+			OAuthList = append(OAuthList, source.ID)
+			app, err := auth.GetOAuth2ApplicationByID(ctx, OAuthList[0])
```

Comment:
Here it looks up an application to get the redirect URL, not an authentication source.